pkg/operator/workloads/data_job.go

```diff
@@ -251,12 +251,7 @@ func dataWorkloadSpecs(ctx *context.Context) ([]*WorkloadSpec, error) {
 		}
 		trainingDatasets = append(trainingDatasets, modelName)
 		trainingDatasetIDs.Add(dataset.GetID())
-		dependencyIDs := ctx.AllComputedResourceDependencies(dataset.GetID())
```

Comment:
I think we still need to append the `transformedColumn.Compute`s since transforming the data happens in the same step as preparing the dataset